### ISSUE TITLE
Compile error with Intel icc compiler

### ISSUE THREAD

**Issue by bqminh** [upsj/terraphast-one#18]
Hlinka Ondrej (CSIRO) got a compilation error with icc (see below). Do you think whether you can support this compiler? Note that the error messages popped up while building IQ-TREE, but the errors occurred while compiling terraphast.

 gcc and clang/llvm versions compile fine, however using the intel compilers (version 17.0.2.174) -  1.7.0beta5 fails to build with the following error (architecture is Intel(R) Xeon(R) CPU E5-2690 v4 which does include sse4_2 capabilities and popcnt:

```
icpc  -DIQTREE_TERRAPHAST -DIQ_TREE -DUSE_BOOSTER -D_USE_PTHREADS -I/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5 -I/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/yaml-cpp/include -I/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/build -I/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/include -I/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib -I/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/intel  -Ofast -fPIC -std=c++11 -fopenmp  -O3 -DNDEBUG   -I/apps/eigen/3.2.9/include/eigen3 -D__SSE3 -msse3 -Wall -Wextra -Wconversion -Wsign-conversion -Werror -std=gnu++11 -o CMakeFiles/terraphast.dir/lib/advanced.cpp.o -c /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/advanced.cpp
In file included from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/bits.hpp(4),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/small_bipartition.hpp(10),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/multitree_iterator.hpp(5),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/advanced.cpp(8):
/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/intel/intrinsics.hpp(21): error: identifier "_popcnt" is undefined
 inline index popcount(index word) { return index(_popcnt(word)); }
                                                  ^

In file included from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/bits.hpp(4),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/small_bipartition.hpp(10),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/multitree_iterator.hpp(5),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/advanced.cpp(8):
/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/intel/intrinsics.hpp(25): error #1682: implicit conversion of a 64-bit integral type to a smaller integral type (potential portability problem)
 	_BitScanForward(&idx, word);
 	                      ^

In file included from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/bits.hpp(4),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/small_bipartition.hpp(10),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/multitree_iterator.hpp(5),
                from /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/advanced.cpp(8):
/home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/intel/intrinsics.hpp(31): error #1682: implicit conversion of a 64-bit integral type to a smaller integral type (potential portability problem)
 	_BitScanReverse(&idx, word);
 	                      ^

compilation aborted for /home/hli001/flush3/installs/iqtree/IQ-TREE-1.7-beta5/terraphast/lib/advanced.cpp (code 2)
```

**Comment by upsj** [comment]
This was due to a non-standard macro used to distinguish between 32 and 64bit systems. Thanks for bringing this to my attention! Should be fixed with 4004c737f

**Comment by bqminh** [comment]
wow that's quick! thanks
